src/Plugin.php

```diff
@@ -110,6 +112,28 @@ public static function init() {
     add_action('wp_enqueue_scripts', __CLASS__ . '::wp_enqueue_scripts');
   }
 
+  /**
+   * Redirects uppercase containing shop slugs to the correct shop page.
+   *
+   * This is necessary as the archives rewrite rules do not match
+   * case-insensitive but match a generic page name instead.
+   *
+   * @implements parse_request
+   */
+  public static function parse_request($request) {
+    if (!$pagename = $request->query_vars['pagename'] ?? '') {
+      return;
+    };
+    $shop_page = get_post(get_option('woocommerce_shop_page_id'));
+    $shop_page_slug = $shop_page->post_name;
```

Comment:
```suggestion
    $shop_page_slug = get_post_field('post_name', get_option('woocommerce_shop_page_id'), 'raw');
```

Reply:
Note that this may not be configured yet when the plugin is installed initially / when woocommerce is installed only afterwards.

Reply:
Good catch, thank you!